sdk/core/azure-core/src/exception.cpp

```diff
@@ -21,25 +21,36 @@ namespace Azure { namespace Core {
 
   RequestFailedException::RequestFailedException(
       std::unique_ptr<Azure::Core::Http::RawResponse>& rawResponse)
-      : RequestFailedException("Received an HTTP unsuccessful status code.")
+      : std::runtime_error(GetRawResponseErrorMessage(rawResponse)),
+        RawResponse(std::move(rawResponse)),
+        // These are guaranteed to always be present in the rawResponse.
+        StatusCode(RawResponse->GetStatusCode()), ReasonPhrase(RawResponse->GetReasonPhrase()),
+        // The response body may or may not have these fields
+        ErrorCode(GetRawResponseField(RawResponse, "code")),
+        Message(GetRawResponseField(RawResponse, "message"))
   {
-    const auto& headers = rawResponse->GetHeaders();
-
-    // These are guaranteed to always be present in the rawResponse.
-    StatusCode = rawResponse->GetStatusCode();
-    ReasonPhrase = rawResponse->GetReasonPhrase();
-    RawResponse = std::move(rawResponse);
-
-    // The response body may or may not have these fields
-    ErrorCode = GetRawResponseField(RawResponse, "code");
-    Message = GetRawResponseField(RawResponse, "message");
-
+    const auto& headers = RawResponse->GetHeaders();
     ClientRequestId = HttpShared::GetHeaderOrEmptyString(headers, HttpShared::MsClientRequestId);
     RequestId = HttpShared::GetHeaderOrEmptyString(headers, HttpShared::MsRequestId);
   }
 
+  std::string RequestFailedException::GetRawResponseErrorMessage(
+      std::unique_ptr<Azure::Core::Http::RawResponse> const& rawResponse)
+  {
+    std::string returnValue("Received an HTTP unsuccessful status code: ");
+    // The status code will always be present in the rawResponse.
+    returnValue += std::to_string(static_cast<int>(rawResponse->GetStatusCode()));
```

Comment:
we might want to translate the status code from int to string. For example from 404 to 'Not Found', or `404 Not Found`

You can get the error description by running:
```cpp
std::string description(
    static_cast<typename std::underlying_type<Azure::Core::Http::HttpStatusCode>::type>(
        rawResponse->GetStatusCode()));
```

Reply:
I thought that `rawResponse->GetReasonPhrase()` returned the error description, is that not the case?

Reply:
The `ResonPhrase` is not equal to the `StatusCode.toString()`  

The status code number and its translation to string is defined in some RFC and we have it in:

https://github.com/Azure/azure-sdk-for-cpp/blob/main/sdk/core/azure-core/inc/azure/core/http/http_status_code.hpp

But, the reason phrase is set by the Server whatever they want.  For example, a Server can return a 404 with reason phrase = "I won't give you that`.  And that's a valid response.

We have the RFC name of each code and we can use it for, as you said before, people don't need to learn the integer codes

Reply:
Can you help me understand how the quoted line works? As best as I can figure, std::underlying_type<HttpStatusCode> returns "int" and that can't be used to initialize a std::string. I also searched through the code and I can find the RFC names in some recorded JSON files but not in the source code. So I'd *love* to understand how to make this work.

Reply:
:( I just noticed it only gets the `int`.  For some reason I thought we had that for getting the string.  Ignore me then :)

Reply:
:(. This makes me sad. But so be it.